### ISSUE TITLE
Desktop - Console error when opening the app

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#12492]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open app
2. Check the console

## Expected Result:
Should be no console error

## Actual Result:
Error in console

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Desktop App

**Version Number:** 1.2.24.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** No

**Email or phone of affected tester (no customers):** 

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/200095103-796a1c3c-c81c-49e4-9ca1-9fe40c284123.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by Santhosh-Sellavel** [comment]
@kbecciv 
Note: This happens on production as well
https://github.com/Expensify/App/pull/12296 is pending production deployment
<img width="1190" alt="Screenshot 2022-11-08 at 4 19 03 AM" src="https://user-images.githubusercontent.com/85645967/200431908-d52ffcb1-d0b3-406f-87f7-89ee7ec66c76.png">

**Comment by flaviadefaria** [comment]
Adding the external label. I'll create the UW post tomorrow.

**Comment by smrutiparida** [comment]
The issue seems to be introduced in Electron 20 https://github.com/electron/forge/issues/2931

**Comment by flaviadefaria** [comment]
@kbecciv @Luke9389 what are your thoughts on the solution @smrutiparida posted above?

**Comment by Santhosh-Sellavel** [comment]
Why do we need to set nodeIntegration: true, what does it do other than remove the warning?

**Comment by smrutiparida** [comment]
> Why do we need to set nodeIntegration: true, what does it do other than remove the warning?

Starting from Electron 20, the sandbox is enabled by default for renderer processes. When we run mainWindow as a separate process, the process is sandboxed. The sandbox limits the harm that malicious code can cause by limiting access to most system resources — sandboxed processes can only freely use CPU cycles and memory. 

However running a preload script contextBridge.js require access to nodeJS modules from the javascript code. This was not a concern since sandbox used to be false by default for Electron <= 20

Starting from Electron 20, by doing nodeIntegration: true we set sandbox: false.  I assumed this is safe which I after reading further thinks are not and need to be well tested. The APIs available in sandbox option to the preload script are more limited than nodeIntegration. Hence instead, I suggest only do sandbox:false now.

sandbox:false shall enable the Node.js engine and make NodeJS's features such as NodeJS's implementation of module, exports, and require to be accessible directly from the HTML/Javascript page's script context.

The Electron FAQ here explains it https://www.electronjs.org/docs/latest/faq#i-can-not-use-jqueryrequirejsmeteorangularjs-in-electron
More - https://www.electronjs.org/docs/latest/tutorial/sandbox#disabling-the-sandbox-for-a-single-process
https://www.electronjs.org/docs/latest/api/browser-window
https://github.com/electron/forge/commit/6e0a6248

**Comment by Santhosh-Sellavel** [comment]
>  I suggest only do sandbox:false now.

Seems straightforward, so looks good to me! Please have a look at this when you have time @Luke9389!

**Comment by puneetlath** [comment]
@Luke9389 this is one of the oldest issues in the /App repo. To help us clear out the large backlog of bugs, can you:  
- Decide whether any proposals currently meet our guidelines and can be approved as-is 
- For any that can't, please take this issue internal and treat it as one of your highest priorities 
- If you have any questions, don't hesitate to start a discussion in #bug-zero

**Comment by flaviadefaria** [comment]
Internal post -https://www.upwork.com/ab/applicants/1593608404131282944/job-details
External post - https://www.upwork.com/jobs/~01c43f9dac33b0431c

**Comment by d-mykhailo** [comment]
**Proposal:**

In Electron 20 default renderer is sandbox.
As preload scripts are depending on Node, you need to enable node in workers.
with nodeIntegrationInWorker: true, you can do that like this. (desktop/main.js/mainWindow)

    `const browserWindow = new BrowserWindow({
                backgroundColor: '#FAFAFA',
                width: 1200,
                height: 900,
                webPreferences: {
                    preload: `${__dirname}/contextBridge.js`,
                    contextIsolation: true,
                    nodeIntegrationInWorker: `true,
                },
                titleBarStyle: 'hidden',
            });`

@flaviadefaria , please check

**Comment by Santhosh-Sellavel** [comment]
> > I suggest only do sandbox:false now.
> 
> Seems straightforward, so looks good to me! Please have a look at this when you have time @Luke9389!

bump @Luke9389

**Comment by Luke9389** [comment]
Checking this out now everyone.

**Comment by Luke9389** [comment]
Yep, let's move forward with this solution: https://github.com/Expensify/App/issues/12492#issuecomment-1309838065

Thanks for the patience everyone.

**Comment by smrutiparida** [comment]
I will have the PR soon. Currently I am testing it locally.

**Comment by flaviadefaria** [comment]
@smrutiparida do you have an update?

**Comment by smrutiparida** [comment]
Not yet. Need a day more.

**Comment by Santhosh-Sellavel** [comment]
Seems like a simple and straightforward change, can you let us know what's causing the delay?

**Comment by smrutiparida** [comment]
I need to create the screenshots in all devices of the errors not coming along with iOS. which is taking some time

**Comment by Santhosh-Sellavel** [comment]
Again we are specifically modifying desktop-only code, so I think we don't need it for other platforms, only on this issue we didn't touch native src code. Please include that in your PR description.

**Comment by iwiznia** [comment]
Sorry, not very clear, did we hire @smrutiparida? If so, can someone assign them so the help wanted label is removed?

**Comment by Santhosh-Sellavel** [comment]
@flaviadefaria please assign @smrutiparida issue.

@smrutiparida Any update on the PR?

**Comment by flaviadefaria** [comment]
It seems like the PR is still being reviewed.

**Comment by smrutiparida** [comment]
I have applied in upwork for the job.

**Comment by Santhosh-Sellavel** [comment]
PR I'd merged

**Comment by flaviadefaria** [comment]
Heads up I'm heading OoO for a week so in case this needs to be paid out before I come back then please reassign or ask someone else in the team to process the payment. Thanks!

**Comment by JmillsExpensify** [comment]
Still working on getting the linked PR deployed.

**Comment by flaviadefaria** [comment]
It looks like the [PR was deployed](https://github.com/Expensify/App/pull/13045#issuecomment-1349934516) so I'll proceed to paying the contributors. Let me know in case I'm missing something.

**Comment by flaviadefaria** [comment]
Offers sent in UW!

**Comment by Luke9389** [comment]
Thanks @flaviadefaria

**Comment by flaviadefaria** [comment]
Contributors paid, thanks!
